### PR TITLE
Made the code more modular, and fixed an issue with html.h 

### PR DESCRIPTION
**Pull request by frags51**
#72 : Split the file "bric.c" into multiple files under the src/ directory. Also, some changes to the way the modules/syntax/lang/lang.h files are written. bric.c now only contains some tag related functions and editor init/processing functions.
**The build is running perfectly and is mergeable with the master branch as of now.**

Also, there was a minor error in html.h (no NULL termination in html_keywords).

### CODE REVIEW

**Comment by shnupta** [comment]
Hi there,

I think changing all the languages to .c files rather than .h files isn't really needed for now and it'll cause a lot more trouble in the long run.

Apart from that the rest of your work looks good, but would you possibly be able to separate them into different PRs that are slightly more specific so they are easier to manage.

Thanks :)

**Comment by frags51** [comment]
Hi
I am busy for the next few weeks, so I won't have time to change the commits. So won't be able to help you there.
Also, what trouble do you think would be caused by the '.c' files? '.h' files should contain declarations only, not definitions. Multiple includes cause compilation errors, if .h files contain definitions, as happened in this case too. That's why I separated definitions into the '.c' files.

File: modules/syntax/syntax.h [hunk 1]
```diff
@@ -44,26 +44,9 @@ typedef struct hlcolour {
...
-  RUBY_syntax,
-  RUST_syntax,
-  HTML_syntax,
-  D_syntax,
-  BRAIN_syntax,
-  BASH_syntax,
-  MAKEFILE_syntax
-};
-
-#define HIGHLIGHT_DB_ENTRIES (sizeof(highlight_db)/sizeof(highlight_db[0]))
```
Comment (gaurav-git) [inline modules/syntax/syntax.h]: I can see you have moved lots of definitions in C source file, I hope you have not moved those which are common to multiple source file. Let's say for example if in future it is required to use `struct editor_syntax highlight_db[]` to define a variable in one of the source file then one needs to declare the struct as `extern` in that source file.
I would like suggestions from others as well on this. @shnupta what do you think on this?

File: bric.h [hunk 2]
```diff
@@ -31,208 +31,31 @@
 #define SELECTION_MODE 1
 #define NORMAL_MODE 2
```
Comment (gaurav-git) [inline bric.h]: In my opinion it would be better to use `enum` instead of `macros`.

**Comment by frags51** [comment]
@gaurav-git : **1** Didn't exactly understand what you mean. What is the problem with using extern? I tried to modify as less as possible. Moving those definitions was necessary, as I explained earlier, to prevent initialization of variables in header files, which are included multiple times leading to multiple initializations (and hence compilation errors). I believe this is how code should be: declarations in '.h' files and definitions in '.c' files. Keeping this in mind, any future changes **won't** lead to any confusions.

**2** I agree. Enums do provide a safer code (in the sense that you are less likely to make errors while using enums), while macros provide (very very) slight performance advantage. I guess it depends on the programmer what he prefers.  However, I too recommend using enums in this case.

**Comment by gaurav-git** [comment]
@frags51 As as far as variables are concerned I am ok moving them to *.c source files but let the structure, enum's and union reside in the header file.

**Comment by frags51** [comment]
I don't think I have moved structures to '.c' files.(It's been 3 weeks since I worked on this :P) Can you please give a reference to the part of whatever file you are talking about?

**Comment by gaurav-git** [comment]
One I can see is `struct editor_syntax highlight_db` from syntax.h to syntax.c. 
It seems that there are others as well, please restore the struct's, enums etc to the header. Apart from that everything looks okay.

**Comment by frags51** [comment]
@gaurav-git The one you are talking about is **NOT** a struct, rather it is an **array** of editor_syntax structs. Hence, it is an initialization of a variable. Including it in '.h' files would lead to multiple initialization. However, it is declared to be an extern in the syntax.h file, hence any '.c' file that includes syntax.h will be able to use that **array**. I hope the confusion is cleared. I am sure I have not moved any other unnecessary entities to '.c' files.

**Comment by gaurav-git** [comment]
@frags51 Alright buddy. I missed that, my bad.

**Comment by shnupta** [comment]
Hi @frags51 ,

I think moving these to .c files is now probably a good idea and will actually make it easier in future in case we need to restructure the editor etc.

Whenever you have time next, do you think you could create separate pull requests for each of the changes e.g. 'syntax module changes' and 'bric.c modular changes' for example.

Many thanks :)

**Comment by frags51** [comment]
@shnupta I have tried to make the commits more sensible. Doing even smaller commits would really require reworking, since bric.c is different at each stage and I only have the finished file. I wouldn't have time to do that. Hope this is sufficient.

**Pull request closed without merging**